src/main/java/homer/model/lights/Light.java

```diff
@@ -103,8 +108,18 @@ public void setIstantConsumption(final double istantConsumption) {
     }
 
     @Override
-    public void plug(Outlet outlet) {
-        this.outlet = outlet;
+    public void plug(final Outlet outlet) {
+        this.outlet = new Outlet(outlet.getInfo(), outlet.getState(), outlet.getMinValue(), outlet.getMaxValue());
+    }
+
+    /**
+     * Returns the {@link homer.model.outlets.Outlet} the device is plugged to.
+     * 
+     * @return outlet.
+     */
+    public Outlet getOutlet() {
+        return new Outlet(this.outlet.getInfo(), this.outlet.getState(), this.outlet.getMinValue(),
+                this.outlet.getMaxValue());
```

Comment:
I had to create a defensive copy of the outlet, should I create a new constructor for Outlet which takes a Outlet as input parameter and sets the values like I did here?

Reply:
yes, i don't really see the point of doing it like this

Reply:
Yep, as I said, it was because of spotbugs, you must return a defensive copy of the object. I'll implement the new constructor for Outlet and change this, it's still wip.